### PR TITLE
Pre release fixes

### PR DESCRIPTION
**Pull request by sergeysikachev**
- Developer(s): @sergeysikachev

- Peer Reviewer:

- Ticket: URL_HERE

- Project PR: URL_HERE

- Academy PR: URL_HERE (for feature documentation and if needed migration guides and project integration guide)

#### Please confirm

- [ ] No new OS components - or they have been approved by legal department
- [ ] All new or heavily changed classes get an "A" rating (Scrutinizer), except Facades, Factories, QueryContainers and Tests

#### Documentation

- [ ] Functional documentation provided or in progress?
- [ ] Integration guide for projects provided or not needed?
- [ ] Migration guides for all contained majors provided or not needed?

#### Release Table

   Module                | Release Type         | Constraints         |
   :--------------------- | :------------------------ | :--------------------- |
   Payone               |       patch                 |                       |

#### Release Notes

- Integrate payone eco with master branch of demoshop 
- Update composer dependencies versions
- Refactoring method names etc.

-----------------------------------------

#### Bundle Payone

_**Major:** Backwards-compatible bug fix_

Def of done (by responsible developer):
- [ ] All changes are backward-compatible. Outdated code is marked as deprecated
- [ ] The change is isolated and not mixed with any cleanups or other changes
- [ ] New code fits to the architecture rules
- [ ] All new or changed business logic is covered by functional tests (in Zed business layer)

##### Change log

Bugfixes

- Fixed compatibility with the latest demoshop.

### CODE REVIEW

**Comment by sergeysikachev** [comment]
Probably it should be released as version 0.1.1.

File: src/SprykerEco/Client/Payone/Zed/PayoneStubInterface.php [hunk 1]
```diff
@@ -0,0 +1,86 @@
...
+    public function initPaypalExpressCheckout(PayoneInitPaypalExpressCheckoutRequestTransfer $requestTransfer);
+
+    /**
+     * @param \Generated\Shared\Transfer\QuoteTransfer $quoteTransfer
+     *
+     * @return \Generated\Shared\Transfer\PayonePaypalExpressCheckoutGenericPaymentResponseTransfer
+     */
+    public function getPaypalExpressCheckoutDetails(QuoteTransfer $quoteTransfer);
+
+}
```
Comment (dereuromark) [inline src/SprykerEco/Client/Payone/Zed/PayoneStubInterface.php]: careful about the missing lineendings.

File: src/SprykerEco/Yves/Payone/Controller/ExpressCheckoutController.php [hunk 2]
```diff
@@ -37,7 +37,7 @@ public function initPaypalExpressCheckoutAction()
     }
 
     /**
-     * @return \Symfony\Component\HttpFoundation\RedirectResponse
+     * @return \Symfony\Component\HttpFoundation\RedirectResponse | array
```
Comment (dereuromark) [inline src/SprykerEco/Yves/Payone/Controller/ExpressCheckoutController.php]: spacing around pipes is invalid

File: src/SprykerEco/Zed/Payone/Business/Api/Response/Container/GenericPaymentResponseContainer.php [hunk 3]
```diff
@@ -301,7 +306,7 @@ public function setEmail(string $email)
      */
     protected function getPreparedKey($key)
     {
-        $key = preg_replace("/add_paydata\[(.*)\]/", "$1", $key);
+        $key = preg_replace(self::ADD_PAYDATA_PATTERN, "$1", $key);
```
Comment (dereuromark) [inline src/SprykerEco/Zed/Payone/Business/Api/Response/Container/GenericPaymentResponseContainer.php]: not sure if the const is helping here.

**Comment by dereuromark** [comment]
Make it a patch in the release template if you want this to be a patch 0.1.1
Looks good apart from my comments :+1:

File: src/SprykerEco/Zed/Payone/Business/Api/Response/Container/GenericPaymentResponseContainer.php (hunk 3 above)
Comment (sergeysikachev) [inline src/SprykerEco/Zed/Payone/Business/Api/Response/Container/GenericPaymentResponseContainer.php]: It was a code review comment from Alex Galych, he wanted this to be moved to constant. I agree, without the const it is more clear, I'll revert this change.

**Comment by dereuromark** [comment]
Please make sure you set up the code sniffer and alike for this standalone repo to hold the code quality high.
Some of the issues above would be detected here.

**Comment by sergeysikachev** [comment]
Yes, first I will set up phpcs and phpmd to be run on precommit git hook to avoid fixing phpcs issues after it is committed.
I have changed the type to patch in the description. Could you please make it a release 0.1.1 ?

**Comment by dereuromark** [comment]
Sure, once approval from PS (sergey or alex) is here (thumps up) I will do it asap.

**Comment by sbraznikov** [comment]
👍

**Pull request merged**